test/y2storage/encryption_processes/luks2_pervasive_test.rb

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env rspec
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../spec_helper"
+require "y2storage"
+
+describe Y2Storage::EncryptionProcesses::Pervasive do
+  subject(:process) { described_class.new(method) }
+  let(:method) { double }
+
+  describe ".used_for?" do
```

Comment:
Since this is quite internal thing, we are so far testing this from a more external entry point here
https://github.com/yast/yast-storage-ng/blob/master/test/y2storage/encryption_method_test.rb#L193

I find the alternative in this pull request to be basically a fully mocked repetition of the implementation.

I think I would prefer a couple of contexts in the above-mentioned test of `EncryptionMethod.for_device`.